src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,181 @@
+    return null;
+  }
+
+  useEffect(() => {
+    /* Ensure correct layer is active on app load */
+    if (currentLayer !== getVisibleLayer(baseLayers)) {
+      setCurrentLayer(getVisibleLayer(baseLayers));
+    }
+  }, [currentLayer]);
+
+  useEffect(() => {
+    /* Used for correct layer image render with animation */
+    if (!switcherOpen) {
+      const timeout = setTimeout(() => {
+        setIsClosed(true);
+      }, 200);
+      return () => clearTimeout(timeout);
+    }
+    return setIsClosed(false);
+  }, [switcherOpen]);
+
+  const toggleBtn =
+    baseLayers.length > 0 ? (
+      <div
+        className="rs-base-layer-switch-close"
+        role="button"
+        onClick={() => setSwitcherOpen(false)}
+        onKeyPress={e => e.which === 13 && setSwitcherOpen(false)}
+        tabIndex="0"
+        aria-label={titles.closeSwitcher}
+        title={titles.closeSwitcher}
+      >
+        <FaChevronCircleLeft size={15} focusable={false} />
+      </div>
+    ) : null;
+
+  return (
+    <div className={`${className}${switcherOpen ? ' open' : ''}`}>
+      <div className="rs-base-layer-switch-layers">
+        {switcherOpen ? toggleBtn : null}
+        {baseLayers.map((layer, index) => {
+          const layerName = layer.getName();
+          return (
+            <div
+              key={layer.key}
+              className={`rs-base-layer-switch-button${
+                switcherOpen ? ' open' : ''
+              }${layerName === currentLayer.getName() ? ' rs-active' : ''}`}
+              role="button"
+              title={isClosed ? titles.openSwitcher : layerName}
+              aria-label={isClosed ? titles.openSwitcher : layerName}
+              onClick={() => {
+                if (!switcherOpen) {
+                  setSwitcherOpen(true);
+                  return;
+                }
+                setCurrentLayer(layer);
+                layer.setVisible(true);
+                setSwitcherOpen(false);
+              }}
+              onKeyPress={e => {
+                if (e.which === 13) {
+                  if (!switcherOpen) {
+                    setSwitcherOpen(true);
+                  } else {
+                    layer.setVisible(true);
+                    setCurrentLayer(layer);
+                    setSwitcherOpen(false);
+                  }
+                }
+              }}
+              tabIndex={`${isClosed ? index : '0'}`}
+            >
+              <div
+                className={`rs-base-layer-switch-title${
+                  isClosed ? ' closed' : ''
+                }`}
+              >
+                {isClosed ? titles.button : layerName}
+              </div>
+              <img
+                src={
+                  isClosed
+                    ? getNextImage(currentLayer, baseLayers, images)
+                    : images[index]
+                }
+                alt={altText}
+                className="rs-base-layer-switch-image"
+              />
+            </div>
```

Comment:
Maybe toi be more understandable, create to <div>
```
isClosed ?<><div  className={`rs-base-layer-switch-title rs-closed}>titles.button</div><img ...></>
: <><div  className={`rs-base-layer-switch-title}>layerName</div><img ...></>
```

Reply:
More lines of code, but fair enough